avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerdeCache.java

```diff
@@ -296,6 +339,26 @@ private String getSchemaKey(Schema writerSchema, Schema readerSchema) {
         Utils.getSchemaFingerprint(readerSchema));
   }
 
+  /**
+   * A wrapper function that limits the total number of fast de/serilizer calsses generated
+   *
+   * @param runnable The function to build and save fast de/serializer
+   */
+  private void buildFastClassWithNumCheck(Runnable runnable) {
+    try {
+      if (this.generatedSerDesNum.get() < this.generatedFastSerDesLimit) {
+        runnable.run();
+      } else if (this.generatedSerDesNum.get() == this.generatedFastSerDesLimit) {
+        LOGGER.warn("Generated fast serdes classes number hits limit {}", this.generatedFastSerDesLimit);
+      } else if (this.generatedSerDesNum.get() > this.generatedFastSerDesLimit) {
+        LOGGER.debug("Generated serdes number {}, with fast serdes limit set to {}", this.generatedSerDesNum.get(), this.generatedFastSerDesLimit);
+      }
+      this.generatedSerDesNum.incrementAndGet();
```

Comment:
Race condition could happen here since the check and update are happening in different places.
I don't think it is a big concern considering the default thread pool size is 2, so in the worst scenario, the total number of the generated classes could be one more than the specified limit.
But it is good to  call it out here, and we need to use `AtomicInteger` here since there is no difference from an `Integer` according to the actual usage.